pkg/resizer/csi_resizer.go

```diff
@@ -131,14 +145,30 @@ func (r *csiResizer) CanSupport(pv *v1.PersistentVolume) bool {
 	return true
 }
 
+// Resize resizes the persistence volume given request size
+// It supports both CSI volume and migrated in-tree volume
 func (r *csiResizer) Resize(pv *v1.PersistentVolume, requestSize resource.Quantity) (resource.Quantity, bool, error) {
 	oldSize := pv.Spec.Capacity[v1.ResourceStorage]
 
-	source := pv.Spec.CSI
-	if source == nil {
-		return oldSize, false, errors.New("not a CSI volume")
+	var volumeID string
+	if pv.Spec.CSI != nil {
+		// handle CSI volume
+		source := pv.Spec.CSI
+		volumeID = source.VolumeHandle
+	} else {
+		if csitranslationlib.IsMigratedCSIDriverByName(r.name) {
+			// handle migrated in-tree volume
```

Comment:
Can this logic from L154 - L170 be compacted in the following manner (do the translation first if necessary and then simply grab the volumeID from the translated pv.Spec.CSI):

```
if csitranslationlib.IsMigratedCSIDriverByName(r.name) {
	pv, err = csitranslationlib.TranslateInTreePVToCSI(pv); err != nil {
		return oldSize, false, fmt.Errorf("failed to translate persistent volume: %v", err)
	}
}
source := pv.Spec.CSI
if source == nil {
	return oldSize, false, errors.New("cannot process a volume without a CSI source")
}
volumeID := source.VolumeHandle

```

Reply:
Good idea. Updated

Reply:
hmmm. Just tested and find out this won't work for the case of CSI volume since `csitranslationlib.IsMigratedCSIDriverByName(r.name)` returns true and it will attempt to translate a CSI volume to CSI